File: clojure/series/src/series.clj
(ns series)

(defn slices
  "Returns all contiguous substrings of length n from the string s."
  [s n]
  (when (empty? s) 
    (throw (IllegalArgumentException. "series cannot be empty")))
  (when (> n (count s)) 
    (throw (IllegalArgumentException. "slice length cannot be greater than series length")))
  (when (zero? n) 
    (throw (IllegalArgumentException. "slice length cannot be zero")))
  (when (neg? n) 
    (throw (IllegalArgumentException. "slice length cannot be negative")))
  (map #(apply str %) (partition n 1 s)))

File: python/series/series.py
"""Output all contiguous substrings of given length in appearance order"""

def slices(series, length):
    if not series:
        raise ValueError('series cannot be empty')
    if length > len(series):
        raise ValueError('slice length cannot be greater than series length')
    if length == 0:
        raise ValueError('slice length cannot be zero')
    if length < 0:
        raise ValueError('slice length cannot be negative')

    return [series[index:index+length] for index in range(0, len(series) - length + 1)]
